viewing_party/party.py

```diff
@@ -1,23 +1,128 @@
 # ------------- WAVE 1 --------------------
 
-def create_movie(title, genre, rating):
-    pass
+def create_movie(movie_title, genre, rating):
+    new_movie = {}
+    if movie_title == None or genre == None or rating == None:
+        new_movie = None
+        return new_movie
+    else:
+        new_movie = {'title':movie_title,'genre':genre, 'rating':rating}
+        return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    count = 0
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            add_to_watched(user_data,movie)
+            del(user_data["watchlist"][count])
+        count += 1
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0
+    for i in range (0,len(user_data["watched"])):
+        print(user_data["watched"][i]["rating"])
+        sum += user_data["watched"][i]["rating"]
+    if sum == 0:
+        average = 0.0
+        return average
+    else:
+        average = sum/len(user_data["watched"])
+        return average
 
+def get_most_watched_genre(user_data):
+    genre_counts = {}
+    counter = 0
+    if len(user_data['watched']) < 1:
+        return None
+    for i in range(len(user_data["watched"])):
+        genre = user_data["watched"][counter]["genre"]
+        genre_count = genre_counts.get(genre,0)
+        genre_counts[genre] = genre_count + 1
+        counter +=1
+    max_genre = max(genre_counts, key=genre_counts.get)

```

Comment:
Great use of the max method here. If you didn't use the max method what would you do in its place?